src/node/desktop/src/main/desktop-browser-window.ts

```diff
@@ -243,6 +243,15 @@ export class DesktopBrowserWindow extends EventEmitter {
 
     // calling event.preventDefault() prevents navigation
     this.window.webContents.on('will-navigate', (event, url) => {
+      // An iframe might try to force navigation to occur in the main frame;
+      // for example, via target="_top". Detect these scenarios and open the
+      // requested URL in an external window instead.
+      if (event.frame != event.initiator) {
```

Comment:
Do we need to worry about nulls/undefined in `event.frame` or `event.initiator`? Maybe that would be overkill, not sure.

```TypeScript
  if (event.frame && event.initiator && event.frame !== event.initiator) {
```